docs/excel/custom-functions-custom-enums.md

```diff
@@ -0,0 +1,174 @@
+---
+title: Custom enums for custom functions
+description: Create custom enums to use with your custom functions.
+ms.date: 06/24/2025
+ms.localizationpriority: medium
+---
+
+# Create custom enums for your custom functions
+
+:::image type="content" source="../images/custom-functions-custom-enum-autocomplete.png" alt-text="The members of a custom enum displaying in the Excel AutoComplete menu.":::
+
+With custom enums, the members in your enum display in the Excel AutoComplete menu for your custom function. The planets in the preceding screenshot are members of a custom enum. This article describes how to create custom enums and use them as parameters in your custom functions.
+
+## Define the custom enum
+
+Define your enum with the JSDoc tag `@customenum`. The corresponding JSON properties are then autogenerated in your custom function metadata. For more information about JSDoc tags and custom functions, see [Basics of JSDoc tags](custom-functions-json-autogeneration.md#basics-of-jsdoc-tags).
+
+> [!NOTE]
+> Custom enums created with the `@customenum` JSDoc tag are only supported in TypeScript. They are not supported in JavaScript. To use custom enums with JavaScript functions, see [Edit enums in JSON metadata](#edit-enums-in-json-metadata).
+
+The following code snippet shows how to define and use a simple custom enum as a parameter.
+
+```typescript
+/** 
+ * A custom enum with descriptions and tooltips. 
+ * @customenum {string} 
+ */
+enum PLANETS { 
+  /** Mercury is the first planet from the sun. */ 
+  mercury = "Mercury", 
+
+  /** Venus is the second planet from the sun. */ 
+  venus = "Venus", 
+
+  /** Earth is the third planet from the sun. */ 
+  earth = "Earth", 
+} 
+
+/** 
+ * A sample function that uses the custom enum as a parameter.
+ * @customfunction 
+ */ 
+function getPlanets(value: PLANETS): any { 
+  return value; 
+} 
+```
+
+## Edit enums in JSON metadata
```

Comment:
Should this section be in the "Manually create JSON metadata for custom functions" article instead? I think that would help keep this article simple and help the common use case of people just using TypeScript.

Reply:
Ok, I'll move it. Thanks for the feedback!